[Bulk] Broken Link fixes (#878)
MicrosoftDocs/biztalk-docs: biztalk/technical-guides/checklist-measuring-performance-on-hyper-v.md

> | **Measure disk latency on a Hyper-V host operating system** – The best initial indicator of disk performance on a Hyper-V host operating system is obtained by using the “\Logical Disk(*)\Avg. Disk sec/Read” and “\Logical Disk(\*)\Avg. Disk sec/Write” performance monitor counters. These performance monitor counters measure the amount of time that read and write operations take to respond to the operating system. As a general rule of thumb, average response times greater than 15ms are considered sub-optimal. As a general rule of thumb, average response times greater than 15ms are considered sub-optimal. This is based on the typical seek time of a single 7200 RPM disk drive without cache. The use of logical disk versus physical disk performance monitor counters is recommended because Windows applications and services utilize logical drives represented as drive letters wherein the physical disk (LUN) presented to the operating system can be comprised of multiple physical disk drives in a disk array. Use the following rule of thumb when measuring disk latency on the Hyper-V host operating system using the \Logical Disk(\*)\Avg. Disk sec/Read or \Logical Disk(\*)\Avg. Disk sec/Write performance monitor counters:<br /><br /> This is based on the typical seek time of a single 7200 RPM disk drive without cache. The use of logical disk versus physical disk performance monitor counters is recommended because Windows applications and services utilize logical drives represented as drive letters wherein the physical disk (LUN) presented to the operating system can be comprised of multiple physical disk drives in a disk array. Use the following rule of thumb when measuring disk latency on the Hyper-V host operating system using the \Logical Disk(\*)\Avg. Disk sec/Read or \Logical Disk(\*)\Avg. Disk sec/Write performance monitor counters:<br /><br /> -   1ms to 15ms  = Healthy<br />-   15ms to 25ms = Warning or Monitor<br />-   26ms or greater = Critical, performance will be adversely affected \*\*Note:*\*  Physical disks installed on a non-virtualized environment offer better performance than disks accessed through a Hyper-V host operating system. If disk performance is absolutely critical to the overall performance of your application, consider hosting disks on physical hardware only. **Note:**  When evaluating disk I/O performance, ensure that you configure antivirus software to exclude scanning of any disk partitions that are being evaluated. Antivirus scanning introduces overhead that can negatively impact performance and skew test results. <br /><br /> **Measure disk latency on guest operating systems** – Response times of the disks used by the guest operating systems can be measured using the same performance monitor counters used to measure response times of the disks used by the Hyper-V host operating system. | For more information about disk performance analysis, see the following resources:<br /><br /> -   “Performance Overhead of Running SQL Server in Hyper-V” section of “Running SQL Server 2008 in a Hyper-V Environment – Best Practices and Performance Considerations” whitepaper [here](https://go.microsoft.com/fwlink/?LinkId=144622).<br />-   Ruling Out Disk-Bound Problems [here](https://go.microsoft.com/fwlink/?LinkId=120947).<br />-   SQL Server Predeployment I/O Best Practices<br />-   “I/O Bottlenecks” section of the “Troubleshooting Performance Problems in SQL Server 2008” whitepaper available [here](https://go.microsoft.com/fwlink/?LinkID=202135).<br />-   How to Identify a Disk Performance Bottleneck Using the Microsoft Server Performance Advisor (SPA) Tool [here](https://go.microsoft.com/fwlink/?LinkID=98096). |

Reviewer: ```suggestion
| **Measure disk latency on a Hyper-V host operating system** – The best initial indicator of disk performance on a Hyper-V host operating system is obtained by using the “\Logical Disk(*)\Avg. Disk sec/Read” and “\Logical Disk(\*)\Avg. Disk sec/Write” performance monitor counters. These performance monitor counters measure the amount of time that read and write operations take to respond to the operating system. As a general rule of thumb, average response times greater than 15ms are considered sub-optimal. As a general rule of thumb, average response times greater than 15ms are considered sub-optimal. This is based on the typical seek time of a single 7200 RPM disk drive without cache. The use of logical disk versus physical disk performance monitor counters is recommended because Windows applications and services utilize logical drives represented as drive letters wherein the physical disk (LUN) presented to the operating system can be comprised of multiple physical disk drives in a disk array. Use the following rule of thumb when measuring disk latency on the Hyper-V host operating system using the \Logical Disk(\*)\Avg. Disk sec/Read or \Logical Disk(\*)\Avg. Disk sec/Write performance monitor counters:<br /><br /> This is based on the typical seek time of a single 7200 RPM disk drive without cache. The use of logical disk versus physical disk performance monitor counters is recommended because Windows applications and services utilize logical drives represented as drive letters wherein the physical disk (LUN) presented to the operating system can be comprised of multiple physical disk drives in a disk array. Use the following rule of thumb when measuring disk latency on the Hyper-V host operating system using the \Logical Disk(\*)\Avg. Disk sec/Read or \Logical Disk(\*)\Avg. Disk sec/Write performance monitor counters:<br /><br /> -   1ms to 15ms  = Healthy<br />-   15ms to 25ms = Warning or Monitor<br />-   26ms or greater = Critical, performance will be adversely affected \*\*Note:*\*  Physical disks installed on a non-virtualized environment offer better performance than disks accessed through a Hyper-V host operating system. If disk performance is absolutely critical to the overall performance of your application, consider hosting disks on physical hardware only. **Note:** When evaluating disk I/O performance, ensure that you configure antivirus software to exclude scanning of any disk partitions that are being evaluated. Antivirus scanning introduces overhead that can negatively impact performance and skew test results. <br /><br /> **Measure disk latency on guest operating systems** – Response times of the disks used by the guest operating systems can be measured using the same performance monitor counters used to measure response times of the disks used by the Hyper-V host operating system. | For more information about disk performance analysis, see the following resources:<br /><br /> - “Performance Overhead of Running SQL Server in Hyper-V” section in the [“Running SQL Server 2008 in a Hyper-V Environment – Best Practices and Performance Considerations” whitepaper](https://go.microsoft.com/fwlink/?LinkId=144622) <br />- [Ruling Out Disk-Bound Problems](https://go.microsoft.com/fwlink/?LinkId=120947) <br />- [SQL Server Predeployment I/O Best Practices](https://seniordba.wordpress.com/2016/04/19/sql-server-io-best-practice/) <br />- “I/O Bottlenecks” section in the [“Troubleshooting Performance Problems in SQL Server 2008” whitepaper](https://go.microsoft.com/fwlink/?LinkID=202135) <br />- [How to Identify a Disk Performance Bottleneck Using the Microsoft Server Performance Advisor (SPA) Tool](https://go.microsoft.com/fwlink/?LinkID=98096) |
```

## Measuring Memory Performance

>  Use the following performance monitor counters to measure the impact of available memory on the performance of a guest operating system installed on a Hyper-V virtual machine:

Reviewer: ```suggestion

Use the following performance monitor counters to measure the impact of available memory on the performance of a guest operating system installed on a Hyper-V virtual machine:
```

|Step|Reference|
|----------|---------------|

> |**Measure available memory on the Hyper-V host operating system** – The amount of physical memory available to the Hyper-V host operating system can be determined by monitoring the “\Memory\Available MBytes” performance monitor counter on the physical computer. This counter reports the amount of free physical memory available to the host operating system. Use the following rules of thumb when evaluating available physical memory available to the host operating system:<br /><br /> <ul><li>**\Memory\Available Mbytes** – Available MBytes measures the amount of physical memory available to processes running on the computer, as a percentage of physical memory installed on the computer. The following guidelines apply when measuring the value of this performance monitor counter:<br /><br /> <ul><li>50% of free memory available or more = Healthy</li><li>25% of free memory available = Monitor</li><li>10% of free memory available = Warning</li><li>Less than 5% of free memory available = Critical, performance will be adversely affected</li></ul></li><li>**\Memory\Pages/sec** – This performance monitor counter measures the rate at which pages are read from or written to disk to resolve hard page faults. To resolve hard page faults, the operating system must swap the contents of memory to disk, which negatively impacts performance. A high number of pages per second in correlation with low available physical memory may indicate a lack of physical memory. The following guidelines apply when measuring the value of this performance monitor counter:<br /><br /> <ul><li>Less than 500 = Healthy</li><li>500 - 1000 = Monitor or Caution</li><li>Greater than 1000 = Critical, performance will be adversely affected</li></ul></li></ul><br /> **Measure available memory on the guest operating system** – Memory that is available to the guest operating systems can be measured with the same performance monitor counters used to measure memory available to the Hyper-V host operating system.|For more information about the impact of available physical memory on application server performance, see the Exchange Server 2003 Help topic [“Ruling Out Memory-Bound Problems”](https://go.microsoft.com/fwlink/?LinkId=121056).|

Reviewer: ```suggestion
|**Measure available memory on the Hyper-V host operating system** – The amount of physical memory available to the Hyper-V host operating system can be determined by monitoring the “\Memory\Available MBytes” performance monitor counter on the physical computer. This counter reports the amount of free physical memory available to the host operating system. Use the following rules of thumb when evaluating available physical memory available to the host operating system:<br /><br /> <ul><li>**\Memory\Available Mbytes** – Available MBytes measures the amount of physical memory available to processes running on the computer, as a percentage of physical memory installed on the computer. The following guidelines apply when measuring the value of this performance monitor counter:<br /><br /> <ul><li>50% of free memory available or more = Healthy</li><li>25% of free memory available = Monitor</li><li>10% of free memory available = Warning</li><li>Less than 5% of free memory available = Critical, performance will be adversely affected</li></ul></li><li>**\Memory\Pages/sec** – This performance monitor counter measures the rate at which pages are read from or written to disk to resolve hard page faults. To resolve hard page faults, the operating system must swap the contents of memory to disk, which negatively impacts performance. A high number of pages per second in correlation with low available physical memory may indicate a lack of physical memory. The following guidelines apply when measuring the value of this performance monitor counter:<br /><br /> <ul><li>Less than 500 = Healthy</li><li>500 - 1000 = Monitor or Caution</li><li>Greater than 1000 = Critical, performance will be adversely affected</li></ul></li></ul><br /> **Measure available memory on the guest operating system** – Memory that is available to the guest operating systems can be measured with the same performance monitor counters used to measure memory available to the Hyper-V host operating system. | For more information about the impact of available physical memory on application server performance, see the [Exchange Server 2003 article, "Ruling Out Memory-Bound Problems"(https://go.microsoft.com/fwlink/?LinkId=121056). |
```

## Measuring Network Performance

>  Hyper-V allows guest computers to share the same physical network adapter. While this helps to consolidate hardware, take care not to saturate the physical adapter. Use the following methods to ensure the health of the network used by the Hyper-V virtual machines:

Reviewer: ```suggestion

Hyper-V allows guest computers to share the same physical network adapter. While this helps to consolidate hardware, take care not to saturate the physical adapter. Use the following methods to ensure the health of the network used by the Hyper-V virtual machines:
```

 For more information about network performance analysis, see [Chapter 15 - Measuring .NET Application Performance](https://go.microsoft.com/fwlink/?LinkId=121073).

## Measuring Processor Performance

>  Use the following methods to evaluate processor performance on a guest operating system installed on a Hyper-V virtual machine:

Reviewer: ```suggestion

Use the following methods to evaluate processor performance on a guest operating system installed on a Hyper-V virtual machine:
```

    For more information about processor utilization review the following resources:

>   - “How To: Identify Functions causing a High User-mode CPU Bottleneck for Server Applications in a Production Environment” [here](https://go.microsoft.com/fwlink/?LinkID=107047).

Reviewer: Link is dead, so replacing with similar:

```suggestion
  - [Investigating Bottlenecks](/biztalk/technical-guides/investigating-bottlenecks).
```
